library/defrag.c

```diff
@@ -371,31 +377,64 @@ void *eblob_defrag_thread(void *data)
 			continue;
 		}
 
+		pthread_mutex_lock(&b->defrag_state_lock);
 		if (b->want_defrag == EBLOB_DEFRAG_STATE_NOT_STARTED)
 			b->want_defrag = EBLOB_DEFRAG_STATE_DATA_COMPACT;
 
-		eblob_defrag(b);
+		chunks_dir = b->defrag_chunks_dir;
+		b->defrag_chunks_dir = NULL;
+		pthread_mutex_unlock(&b->defrag_state_lock);
+
+		eblob_defrag_in_dir(b, chunks_dir);
+
+		free(chunks_dir);
+
+		pthread_mutex_lock(&b->defrag_state_lock);
 		b->want_defrag = EBLOB_DEFRAG_STATE_NOT_STARTED;
+		pthread_mutex_unlock(&b->defrag_state_lock);
+
 		sleep_time = datasort_next_defrag(b);
 	}
 
 	return NULL;
 }
 
-int eblob_start_defrag_level(struct eblob_backend *b, enum eblob_defrag_state level)
+int eblob_start_defrag_in_dir(struct eblob_backend *b, enum eblob_defrag_state level, const char *chunks_dir)
```

Comment:
Does this function start defrag with specific options once, or it rewrites global options, and next defrags will start with new options implicitly?

What is the current scenario: is `eblob_start_defrag` function now deprecated, and should we only use the new one in new code? Is any comment required? If we rewrite options, the new function take affect on the behavior of the old one.

Reply:
This function triggers background defrag with specified parameters. This parameters will be used only with the next (triggered) defrag. `eblob_start_defrag()` is function with default parameters.